mllib/src/main/scala/org/apache/spark/ml/feature/QuantileDiscretizer.scala

```diff
@@ -103,6 +103,13 @@ final class QuantileDiscretizer(override val uid: String)
 
 @Since("1.6.0")
 object QuantileDiscretizer extends DefaultParamsReadable[QuantileDiscretizer] with Logging {
+
+  /**
+   * Minimum number of samples required for finding splits, regardless of number of bins.  If
+   * the dataset has less rows than this value, the entire dataset column will be used.
+   */
+  val minSamplesRequired: Int = 10000
```

Comment:
I don't think you're missing anything.  It's my first time contributing and I just want to be explicit for reviewers of the patch.  I agree that 10K isn't that big, especially out "in the wild".  However, I wasn't sure if there were standards for time/memory consumption for tests so I added the line note so that reviewers with more experience would be aware in case there are established/de facto testing standards.

I'll make the "->" changes you've indicated and push a new commit sometime today.  Thanks.